test/threads.jl

```diff
@@ -141,3 +141,41 @@ end
 @test_throws TypeError Atomic{Bool}
 @test_throws TypeError Atomic{BigInt}
 @test_throws TypeError Atomic{Float64}
+
+# Test atomic fences
+type CommBuf
+    barrier::Atomic{Int}
+    data::Vector{Float64}
+    correct::Bool
+    CommBuf() = new(Atomic{Int}(0), Vector{Float64}(100), false)
+end
+function test_fence_write(commbuf::CommBuf)
+    sleep(0.1)
+    for i in eachindex(commbuf.data)
+        commbuf.data[i] = i
+    end
+    atomic_fence()
+    commbuf.barrier[] = 1
```

Comment:
To actually test the fence, I think you want non atomic store here. Otherwise the acquire/release alone should garentee the order in the two threads.

This test (with the `atomic_fence()` removed) should be a good test for the store and load schematics though.